### PR TITLE
Add detected message when GPS found

### PR DESCRIPTION
**Pull request by peterbarker**
Master:

image

after this PR:

image

And the most common uBlox case where the thing got auto-configured:
image

### CODE REVIEW

File: libraries/SITL/SITL.cpp [hunk 1]
```diff
@@ -499,7 +499,7 @@ const AP_Param::GroupInfo SIM::var_gps[] = {
     // @Param: GPS_TYPE
     // @DisplayName: GPS 1 type
     // @Description: Sets the type of simulation used for GPS 1
-    // @Values: 0:None, 1:UBlox, 5:NMEA, 6:SBP, 7:File, 8:Nova, 9:SBP, 10:Trimble, 19:MSP
+    // @Values: 0:None, 1:UBlox, 5:NMEA, 6:SBP, 7:File, 8:Nova, 9:SBP2, 11:Trimble, 19:MSP
```
Comment (khancyr) [inline libraries/SITL/SITL.cpp:502]: isn't trimble a SBF type ? so we that would be 10 ? @Ryanf55

**Review by khancyr — APPROVED** [review]
Yep, that is less confusing I think

File: libraries/SITL/SITL.cpp (hunk 1 above)
Comment (Ryanf55) [reply libraries/SITL/SITL.cpp:502]: No, Trimble PX1 uses GSOF as a protocol and has no affiliation with SBF.

File: libraries/AP_GPS/AP_GPS.cpp [hunk 2]
```diff
@@ -999,6 +999,13 @@ void AP_GPS::update_instance(uint8_t instance)
             tnow = state[instance].last_corrected_gps_time_us/1000U;
             state[instance].corrected_timestamp_updated = false;
         }
+
+        // we set delta_time_ms to the timeout value when clearing
+        // state; use it being zero to mark first message
+        if (timing[instance].delta_time_ms == GPS_TIMEOUT_MS) {
```
Comment (tridge) [inline libraries/AP_GPS/AP_GPS.cpp:1005]: this seems odd - what stops delta_time_ms from being GPS_TIMEOUT_MS later in the flight?
maybe a "bool announced_gps" variable would be better?

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_GPS/AP_GPS.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_GPS/AP_GPS.cpp]: Added a bool.

File: libraries/SITL/SITL.cpp (hunk 1 above)
Comment (khancyr) [reply libraries/SITL/SITL.cpp:502]: Ok so some comment on ap_gps are misleading then as they said sbf and trimble are similar

**Pull request merged**